pkg/cri/sbserver/container_create.go

```diff
@@ -355,6 +356,9 @@ func (c *criService) volumeMounts(containerRootDir string, criMounts []*runtime.
 		}
 		volumeID := util.GenerateID()
 		src := filepath.Join(containerRootDir, "volumes", volumeID)
+		if !filepath.IsAbs(dst) && goruntime.GOOS != "windows" {
```

Comment:
This will most likely break any runtime that is capable of running containers for an OS different that the one that `containerd` is running on. For example, Windows is capable of running both Windows containers (WCOW) and Linux containers (LCOW). For Linux containers, `/C:/test` is a valid path. For Windows containers, it is not. Other runtimes may be written in the future,  that can run Windows containers on Linux. Any runtime that leverage virtual machines are potentially capable of this.

Paths should be handled based on the sandbox target `OS`, not on `runtime.GOOS`.

The  `volumeMounts()` function ends up being [called in `CreateContainer()`](https://github.com/containerd/containerd/blob/main/pkg/cri/sbserver/container_create.go#L154) which passes the mounts returned by this function to `buildContainerSpec()`. In `buildContainerSpec()` [we check `platform.OS`](https://github.com/containerd/containerd/blob/main/pkg/cri/sbserver/container_create.go#L522-L573) before building the spec.

This would be a better place to check the volume `dst` path and decide to normalize it if the target OS is `Unix`.

Reply:
Good point. It will be correct to add the platform as argument in `volumeMounts` for `sbserver` package.

Reply:
Ohh, great points. Added the `platform` param to sbserver as @fuweid. Thanks!